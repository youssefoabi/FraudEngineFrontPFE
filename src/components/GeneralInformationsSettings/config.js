// eslint-disable-next-line import/prefer-default-export
export const config = [
  {
    columns: 4,
    id: 'name',
    type: 'input',
    label: 'Nom du moteur',
  },
  {
    columns: 4,
    id: 'isEnable',
    type: 'switch',
    defaultValue: false,
    label: 'Moteur active',
  },
  {
    columns: 4,
    id: 'id',
    type: 'input',
    label: 'ID Moteur',
    inputType: 'number',
  },
];
