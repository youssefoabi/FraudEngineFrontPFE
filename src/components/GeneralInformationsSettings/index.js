import { DataEdit, ExpansionPanel, GridItem } from '@cdiscount/backoffice-ui';
import { func } from 'prop-types';
import { Grid } from '@material-ui/core';
import React from 'react';

import { config } from './config';
import { enginePropTypes } from '../EngineForm/enginePropTypes';

const GeneralInformationsSettings = ({ engine, handleChange }) => (
  <ExpansionPanel defaultExpanded title="Informations Générales">
    <Grid container direction="column">
      <GridItem title="Moteur de Recherche">
        <Grid item xs={12} md={12} lg={12}>
          <DataEdit fields={config} values={engine} onChange={handleChange} />
        </Grid>
      </GridItem>
    </Grid>
  </ExpansionPanel>
);

GeneralInformationsSettings.propTypes = {
  handleChange: func,
  engine: enginePropTypes,
};

GeneralInformationsSettings.defaultProps = {
  handleChange: Function.prototype,
  engine: null,
};

export default GeneralInformationsSettings;
