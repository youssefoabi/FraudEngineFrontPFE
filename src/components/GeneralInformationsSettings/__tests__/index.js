import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import GeneralInformationsSettings from '../';

describe(GeneralInformationsSettings.name, () => {
  itRendersCorrectly(<GeneralInformationsSettings />);
});
