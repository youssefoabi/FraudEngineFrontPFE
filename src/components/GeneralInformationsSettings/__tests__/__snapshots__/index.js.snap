// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralInformationsSettings renders correctly 1`] = `
<p
  defaultExpanded={true}
  title="Informations Générales"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="column"
  >
    <h
      fullheight={false}
      title="Moteur de Recherche"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={12}
        md={12}
        xs={12}
      >
        <f
          config={null}
          fields={
            Array [
              Object {
                "columns": 4,
                "id": "name",
                "label": "Nom du moteur",
                "type": "input",
              },
              Object {
                "columns": 4,
                "defaultValue": false,
                "id": "isEnable",
                "label": "Moteur active",
                "type": "switch",
              },
              Object {
                "columns": 4,
                "id": "id",
                "inputType": "number",
                "label": "ID Moteur",
                "type": "input",
              },
            ]
          }
          inputs={null}
          onChange={[Function]}
          onInit={[Function]}
          title={null}
          values={null}
        />
      </WithStyles(ForwardRef(Grid))>
    </h>
  </WithStyles(ForwardRef(Grid))>
</p>
`;
